src/stats.jl

```diff
@@ -451,6 +468,10 @@ Mean(T::Type{<:Number} = Float64; weight = EqualWeight()) = Mean(zero(T), weight
 function _fit!(o::Mean{T}, x) where {T}
     o.μ = smooth(o.μ, x, o.weight(o.n += 1))
 end
+function _fit!(o::Mean{T}, y, n) where {T}
+    o.n += n
+    o.μ = smooth(o.μ, y, n / o.n)
+end
```

Comment:
This won't be correct for `Mean{T,W}` when `W` is something other than `EqualWeight`

Reply:
Does this imply there is also a problem with `_merge!(o::Mean, o2::Mean)`? My thinking is along the lines that fitting `y` `k` times is equivalent to merging two means where the second one has `k` values and a mean of `y`. What is the appropriate way to deal with weights? Like this?

```julia
function _fit!(o::Mean{T}, y, n) where {T}
    o.n += n
    o.μ = smooth(o.μ, y, o.weight(n / o.n))
end
```

Reply:
Never mind, in testing that doesn't work. I can define a special version for `Mean` with equal weight, and then use the generic fallback otherwise